MainRepository/Program.cs

```diff
@@ -1,7 +1,96 @@
-﻿internal class Program
-{
-    private static void Main(string[] args)
-    {
-        Console.WriteLine("Hello, World!");
-    }
+﻿namespace List;
+internal class Program {
+   private static void Main () {
+   }
+}
+
+/// <summary>Implementation of list class</summary>
+/// <typeparam name="T">T is type of variables declared in the List</typeparam>
+public class MyList<T> {
+
+   /// <summary>Constructor used to assign values to the list and count</summary>
+   public MyList () {
+      mList = new T[4];
+      mCount = 0;
+   }
+
+   /// <summary>Gets the number of elements in the list</summary>
+   public int Count => mCount;
+
+   /// <summary>Gets the total number of elements in the list</summary>
+   public int Capacity => mList.Length;
+
+   /// <summary>To get the value in the given index and to set an value to the given index</summary>
+   /// <param name="index">The index position to get or set the value</param>
+   /// <returns>The value in the given index position</returns>
+   /// <exception cref="IndexOutOfRangeException">Throws exception when the given index below 0 (or) greater than or equal to the count</exception>
+   public T this[int index] {
+      get {
+         return index < 0 || index >= mCount ? throw new IndexOutOfRangeException ("Index out of valid range") : mList[index];
+      }
+      set {
+         if (index < 0 || index >= mCount) throw new IndexOutOfRangeException ("Index out of valid range");

```

Comment:
This check is repeated both in get and set. So move this to separate method and call here.